EIPS/eip-6789.md

```diff
@@ -0,0 +1,88 @@
+---
+eip: 6789
+title: Rename `gas` to `mana`
+description: This EIP picks up on Vitalik Buterin's (@vbuterin) original proposal in 2015 to rename `gas` to `mana`
+author: Pascal Caversaccio (@pcaversaccio)
+discussions-to: https://ethereum-magicians.org/t/eip-6789-rename-gas-to-mana/13570
+status: Draft
+type: Standards Track
+category: Interface
+created: 2023-03-27
+---
+
+## Abstract
+
+This EIP picks up on Vitalik Buterin's (@vbuterin) original proposal in 2015 to rename `gas` to `mana`.
+
+## Motivation
+
+Vitalik's original motivation is threefold and is still valid today:
+
+- `mana` more accurately reflects the increased environmental friendliness of Proof-of-Stake (PoS),
+- `mana` is generally understood to be ephemeral and non-transferable, and thus represents accurately the characteristic of how `gas` is actually used within Ethereum, and
+- `mana` is generally considered renewable, whilst (natural) gas is considered non-renewable.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+The core term `gas` MUST be renamed to `mana`. This MUST also include renaming the opcode `Gas` to `Mana`.
+
+## Rationale
+
+The underlying rationale for reviving Vitalik's original proposal from 2015 is that we have finally arrived at the age of PoS, and given the roadmap ahead (i.e. "The Surge", "The Scourge", "The Verge", "The Purge", "The Splurge"), I consider this moment as the last opportunity to make such a far-reaching semantic change.
+
+## Backwards Compatibility
+
+This proposal is not backward compatible as it renames the core term `gas`.
+
+## Test Cases
+
+### Example 1
+
+If a transaction requires more `mana` than allowed by the `manaLimit`, it is reverted as an _out-of-mana_ transaction.
+
+### Example 2
+
+A Solidity contract to estimate the used `mana` via the new `manaleft()` syntax (replacing `gasleft()`) for dedicated function calls.
+
+```solidity
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity 0.8.19;
+
+contract ManaMetering {
+    function oldWay() external view returns (string memory, uint256 manaUsed) {
+        string memory hiMom = "Hi Mom, ";
+        string memory missYou = "miss you.";
+        uint256 startMana = manaleft();
```

Comment:
Since we are in the oldWay function I think you wanted to say ``startGas`` instead of ``startMana``.
Both Example are similar.